app/grandchallenge/profiles/migrations/0015_alter_userprofile_receive_notification_emails.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 4.2.11 on 2024-03-13 08:57
+
+from django.db import migrations, models
+
+from grandchallenge.profiles.models import NotificationSubscriptionOptions
+
+
+def migrate_notification_email_preferences(apps, _schema_editor):
+    UserProfile = apps.get_model("profiles", "UserProfile")  # noqa: N806
+    UserProfile.objects.filter(receive_notification_emails="true").update(
+        receive_notification_emails=NotificationSubscriptionOptions.DAILY_SUMMARY
+    )
+    UserProfile.objects.filter(receive_notification_emails="false").update(
+        receive_notification_emails=NotificationSubscriptionOptions.DISABLED
+    )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("profiles", "0014_userprofile_unread_messages_email_last_sent_at"),
+    ]
+
+    operations = [
+        migrations.AlterField(
```

Comment:
Just to double check - have you tested that this migration actually works? I don't know what the consequences of casting a bool field to char are.

Reply:
I tested this, yes, and it worked. Here is what I did: I first spun up the dev environment with the old model field, updated user preferences to have some who opted in and some who opted out, then changed the field, created the migration (and added the extra migration step) and then ran `migrate` and checked that the user preferences were migrated correctly.